flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-elasticsearch/src/main/java/org/apache/flink/cdc/connectors/elasticsearch/serializer/ElasticsearchEventSerializer.java

```diff
@@ -0,0 +1,230 @@
+            default:
+                throw new UnsupportedOperationException("Unsupported Operation " + op);
+        }
+    }
+
+    private Object[] generateUniqueId(RecordData recordData, Schema schema) {
+        List<String> primaryKeys = schema.primaryKeys();
+        return primaryKeys.stream()
+                .map(
+                        primaryKey -> {
+                            Column column =
+                                    schema.getColumns().stream()
+                                            .filter(col -> col.getName().equals(primaryKey))
+                                            .findFirst()
+                                            .orElseThrow(
+                                                    () ->
+                                                            new IllegalStateException(
+                                                                    "Primary key column not found: "
+                                                                            + primaryKey));
+                            int index = schema.getColumns().indexOf(column);
+                            return getFieldValue(recordData, column.getType(), index);
+                        })
+                .toArray();
+    }
+
+    private Object getFieldValue(RecordData recordData, DataType dataType, int index) {
+        switch (dataType.getTypeRoot()) {
+            case BOOLEAN:
+                return recordData.getBoolean(index);
+            case TINYINT:
+                return recordData.getByte(index);
+            case SMALLINT:
+                return recordData.getShort(index);
+            case INTEGER:
+            case DATE:
+            case TIME_WITHOUT_TIME_ZONE:
+                return recordData.getInt(index);
+            case BIGINT:
+                return recordData.getLong(index);
+            case FLOAT:
+                return recordData.getFloat(index);
+            case DOUBLE:
+                return recordData.getDouble(index);
+            case CHAR:
+            case VARCHAR:
+                return recordData.getString(index);
+            default:
+                throw new IllegalArgumentException("Unsupported type: " + dataType);
+        }
+    }
+
+    public Map<String, Object> serializeRecord(
+            RecordData recordData, Schema schema, ZoneId pipelineZoneId) {
+        List<Column> columns = schema.getColumns();
+        Map<String, Object> record = new HashMap<>();
+        Preconditions.checkState(
+                columns.size() == recordData.getArity(),
+                "Column size does not match the data size.");
+
+        for (int i = 0; i < recordData.getArity(); i++) {
+            Column column = columns.get(i);
+            ColumnType columnType = ColumnType.valueOf(column.getType().getTypeRoot().name());
+            ElasticsearchRowConverter.SerializationConverter converter =
```

Comment:
Can we cache TableId and list of converter to avoid creating convert everytime we meet a old TableId?